### ISSUE TITLE
Convert widget to input doesn't preserve type

### ISSUE THREAD

**Issue by romeobuilderotti** [ssitu/ComfyUI_NestedNodeBuilder#19]
TLDR: if widget converted to input after nesting, I can't connect anything to this input

<img width="546" alt="image" src="https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/144384895/ded5fa1d-1e4d-4ce7-9e9f-dd25c81324a5">

<img width="551" alt="image" src="https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/144384895/b9ad87fe-4d2e-4f23-a970-6fbfb7a2998e">

Steps to reproduce:

1. Create KSampler and something else
2. Nest them
3. Convert `seed` (or `steps` or whatever) to input
4. Try to connect external input to it
5. Fail miserably

If I convert it to input *before* nesting, it will work correctly, but it makes the resulting node less reusable. Is there a way to fix this?

Here's a broken workflow for the reference 
<img width="250" alt="image" src="https://github.com/ssitu/ComfyUI_NestedNodeBuilder/assets/144384895/5cd0a12d-ba36-4898-9fd2-c312e00e445a">
